### ISSUE TITLE
Server Side Include (SSI) / Java Script / Accessibility / Document.write

### ISSUE THREAD

**Issue by StdGit** [wet-boew/cdts-sgdc#158]
CDTS is running mainly on 2 servers.

Canada.ca doesn't support SSI
ssl-templates.services.gc.ca support some SSI Directives

SSI is used in 3 different ways:

1-) To choose the environment server
`
		<script src="https://www.canada.ca/etc/designs/canada/cdts/gcweb/v4_0_32c/cdts/compiled/soyutils.js"></script>`

2-) To allow a fall-back with static file when Javascript is disabled
`<!--#if expr = "${HTTP_HOST} != localhost" --><!--#include virtual="/app/cls/WET/gcweb/v4_0_32c/cdts/static/refTop.html" --><!--#endif-->`

3-) To allow the page to be accessible when Javascript is disabled (Skip to main content with tab key) and have a Canada.ca breadcrumb
`<!--#if expr = "${HTTP_HOST} != localhost" --><!--#include virtual="/app/cls/WET/gcweb/v4_0_32c/cdts/static/top-en.html" --><!--#endif-->`

Questions: 

Do we still need Server Side Include (SSI) ?

Do we assume that JavaScript is always enabled ?

If yes, why are we using Canada.ca since it doesn't support SSI Directives ?

If not, should we continue to support Static files?

**Comment by shawnthompson** [comment]
I think you should get rid of the SSI conditions in your sample files.

`<!--#if expr = "${HTTP_HOST} != localhost" -->`

Focus on using the `local` variant `<script src="../compiled/soyutils.js">`

The long debate of supporting no JavaScript has been going on for almost a decade now. I'll leave that up to your team to decide. If you choose to stay with the fallback files, I wouldn't put the sample files on any server (including Canada.ca and Github) that doesn't support SSI includes.

Having said that, a lot of the JS frameworks now are using `noscript` element with text stating that JS is need to view this page. CDTS is different, where the content is still there but the template is gone. I think that fails gracefully. But that's my opinion.

**Comment by StdGit** [comment]
I also recommend to remove SSI code for 2 reasons:

1-) Applications could run on a server that doesn't implement SSI

2-) For security reason https://owasp.org/www-community/attacks/Server-Side_Includes_(SSI)_Injection

It means that we will have to hardcode the environment path in samples

Though, we still have to decide what happens when JS is deactivates. (Noscript)

**Comment by StdGit** [comment]
Also we should remove document.write (DW) in the same time
https://github.com/wet-boew/cdts-sgdc/issues/136
https://stackoverflow.com/questions/802854/why-is-document-write-considered-a-bad-practice

**Comment by ahmad-shahid** [comment]
SSI statements have been removed for v4.0.39 release. 
For the document.write() issue, we are tracking that in #55 .
